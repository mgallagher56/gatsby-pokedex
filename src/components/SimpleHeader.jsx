import React from 'react'
import { Link } from 'gatsby'

import { makeStyles } from '@material-ui/core/styles'
import { AppBar, Button, Toolbar, Typography } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
  appBar: {
    width: '100%',
    paddingBottom: theme.spacing(2),
    paddingTop: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },
  title: {
    textTransform: 'capitalize'
  },
  button: {
    marginRight: theme.spacing(8)
  }
}))

const SimplifiedHeader = (props) => {
  const { appBar, title, button } = useStyles()

  return (
    <AppBar position='relative' className={appBar}>
      <Toolbar>
        {props.btnText
          ? <Link to='/'>
            <Button
              variant='contained'
              color='secondary'
              size='small'
              className={button}
            >
              {props.btnText}
            </Button>
            </Link>
          : null}
        <Typography variant='h5' component='h1' className={title}>
          {props.title}
        </Typography>
      </Toolbar>
    </AppBar>
  )
}

export default SimplifiedHeader
