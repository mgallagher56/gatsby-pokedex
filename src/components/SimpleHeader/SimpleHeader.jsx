import React from 'react'
import { Link } from 'gatsby'

import { makeStyles } from '@material-ui/core/styles'
import { AppBar, Button, Toolbar, Typography } from '@material-ui/core'

// styles
const useStyles = makeStyles((theme) => ({
  appBar: {
    width: '100%',
    paddingBottom: theme.spacing(2),
    paddingTop: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },
  title: {
    textTransform: 'capitalize'
  },
  button: {
    marginRight: theme.spacing(8)
  }
}))

/**
 * SimpleHeader renders a page header
 *
 * Props:
 *
 * title - adds page title to header
 */

// markup
const SimpleHeader = (props) => {
  const { appBar, title, button } = useStyles()

  return (
    <AppBar position='relative' className={appBar}>
      <Toolbar>
        {props.btnText
          ? <Link to='/'>
            <Button
              data-testid='header-button'
              variant='contained'
              color='default'
              size='small'
              className={button}
            >
              {props.btnText}
            </Button>
          </Link>
          : null}
        {props.title
          ? <Typography
              data-testid='header-title'
              variant='h5' component='h1'
              className={title}
            >
            {props.title}
          </Typography>
          : null}
      </Toolbar>
    </AppBar>
  )
}

export default SimpleHeader
