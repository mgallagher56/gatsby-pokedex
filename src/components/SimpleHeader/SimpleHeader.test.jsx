import React from 'react'
import { render, cleanup } from '@testing-library/react'
import SimpleHeader from './SimpleHeader'

describe('SimpleHeader Component', function () {
  afterEach(cleanup)

  it('renders SimpleHeader Component', () => {
    render(<SimpleHeader />)
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<SimpleHeader />)
    expect(asFragment(<SimpleHeader />)).toMatchSnapshot()
  })

  it('should have Hello world title', () => {
    const { getByTestId } = render(<SimpleHeader title='Hello world' />)
    expect(getByTestId('header-title')).toMatchInlineSnapshot(`
    <h1
      class="MuiTypography-root makeStyles-title-8 MuiTypography-h5"
      data-testid="header-title"
    >
      Hello world
    </h1>
    `)
  })

  it('should have a title', () => {
    const { queryByTestId } = render(<SimpleHeader title='title text' />)
    expect(queryByTestId('header-title')).toBeTruthy()
  })

  it('should not have a title', () => {
    const { queryByTestId } = render(<SimpleHeader />)
    expect(queryByTestId('header-title')).toBeNull()
  })

  it('should have a button', () => {
    const { queryByTestId } = render(<SimpleHeader btnText='Go back' />)
    expect(queryByTestId('header-button')).toBeTruthy()
  })

  it('should not have a button', () => {
    const { queryByTestId } = render(<SimpleHeader />)
    expect(queryByTestId('header-button')).toBeNull()
  })
})
