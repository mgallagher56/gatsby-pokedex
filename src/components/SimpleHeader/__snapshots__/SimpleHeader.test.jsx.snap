// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleHeader Component should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary makeStyles-appBar-4 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    />
  </header>
</DocumentFragment>
`;
